task_manager/task_manager.hh

```diff
@@ -0,0 +1,323 @@
+            return _tm.get_task_ttl();
+        }
+
+        task_manager::task_map& get_tasks() noexcept {
+            return _tasks;
+        }
+
+        const task_manager::task_map& get_tasks() const noexcept {
+            return _tasks;
+        }
+
+        task_manager::task_map& get_all_tasks() noexcept {
+            return _tm.get_all_tasks();
+        }
+
+        const task_manager::task_map& get_all_tasks() const noexcept {
+            return _tm.get_all_tasks();
+        }
+
+        void register_task(task_ptr task) {
+            _tasks[task->get_status().id] = task;
+            _tm.register_task(task);
+        }
+
+        void unregister_task(task_id id) {
+            if (_tasks.contains(id)) {
+                _tasks.erase(id);
+                _tm.unregister_task(id);
+            }
+        }
+
+        future<> stop() noexcept {
+            return _gate.close().then([this] {
+                _tm.unregister_module(_name);
+            });
+        }
+    private:
+        future<unsigned> get_task_shard(task_id id) const;
+        task_ptr retrieve_task(task_id id);
+        future<task_ptr> wait_task(task_id id);
+        friend future<std::vector<task_id>> task_manager::task::run_on_shards(task_ptr task, module_ptr module);
+    };
+public:
+    task_manager(config cfg, abort_source& as) noexcept
+        : _cfg(std::move(cfg))
+        , _as(as)
+        , _update_task_ttl_action([this] { return update_task_ttl(task_ttl()); })
+        , _task_ttl_observer(_cfg.task_ttl.observe(_update_task_ttl_action.make_observer()))
+    {}
+
+    module_ptr get_module(std::string module) noexcept {
+        auto it = _modules.find(module);
+        if (it != _modules.end()) {
+            return it->second;
+        }
+        return nullptr;
+    }
+
+    const module_ptr get_module(std::string module) const noexcept {
+        auto it = _modules.find(module);
+        if (it != _modules.end()) {
+            return it->second;
+        }
+        return nullptr;
+    }
+
+    task_map& get_all_tasks() noexcept {
+        return _all_tasks;
+    }
+
+    const task_map& get_all_tasks() const noexcept {
+        return _all_tasks;
+    }
+
+    std::vector<std::string> get_modules() const {
+        std::vector<std::string> v;
+        for (auto el : _modules) {
+            v.push_back(el.first);
+        }
+        return v;
+    }
+
+    future<> stop() noexcept;
+
+    static future<task_manager::foreign_task_ptr> lookup_task_on_all_shards(sharded<task_manager>& tm, task_id tid);
+
+protected:
+    abort_source& abort_source() noexcept {
+        return _as;
+    }
+
+    std::chrono::seconds get_task_ttl() const {
+        return std::chrono::seconds(_task_ttl);
+    }
+private:
+    uint32_t task_ttl() const noexcept {
+        return _cfg.task_ttl.get();
+    }
+
+    future<> update_task_ttl(uint32_t task_ttl) noexcept {
+        _task_ttl = task_ttl;
+        return make_ready_future<>();
+    }
```

Comment:
Same here. It would be clearer to open code the function in the lambda passed to _update_task_ttl_action.
Or implement the whole function as a single member like:
```
future<> update_task_ttl() noexcept {
    _task_ttl = _cfg.task_ttl.get();
    return make_ready_future<>();
}
```

and then maybe initialize `_update_task_ttl_action(std::mem_fn(&update_task_ttl))`